pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -141,66 +142,82 @@ func (r *ReconcileUserSignup) Reconcile(request reconcile.Request) (reconcile.Re
 		// Error reading the object - requeue the request.
 		return reconcile.Result{}, err
 	}
+	logger = logger.WithValues("username", userSignup.Spec.Username)
 
-	reqLogger = reqLogger.WithValues("username", instance.Spec.Username)
-	if instance.Labels[toolchainv1alpha1.UserSignupStateLabelKey] == "" {
-		if err := r.setStateLabel(reqLogger, instance, toolchainv1alpha1.UserSignupStateLabelValueNotReady); err != nil {
+	if userSignup.Labels[toolchainv1alpha1.UserSignupStateLabelKey] == "" {
+		if err := r.setStateLabel(logger, userSignup, toolchainv1alpha1.UserSignupStateLabelValueNotReady); err != nil {
 			return reconcile.Result{}, err
 		}
 	}
 
-	banned, err := r.isUserBanned(reqLogger, instance)
+	// (migration for 'complete' usersignups only) set the UserSignupActivationCounterAnnotationKey if it is missing
+	if condition.IsTrueWithReason(userSignup.Status.Conditions, toolchainv1alpha1.UserSignupComplete, "") {
+		if userSignup.Annotations == nil { // if annotations is empty, it's omitted when reading the resource, hence it's nil here.
+			userSignup.Annotations = map[string]string{}
+		}
+		if _, exists := userSignup.Annotations[toolchainv1alpha1.UserSignupActivationCounterAnnotationKey]; !exists {
+			log.Info("setting 'toolchain.dev.openshift.com/activation-counter' on existing active user")
+			userSignup.Annotations[toolchainv1alpha1.UserSignupActivationCounterAnnotationKey] = "1"
```

Comment:
> We should actually set this to 1 only if there is an existing MUR for this UserSignup. 

isn't it what the `if condition.IsTrueWithReason(userSignup.Status.Conditions, toolchainv1alpha1.UserSignupComplete, "") ` statement on line 154 does? (or at least, that was my intent)

Reply:
> And let's create a JIRA to delete this migration function as soon as we done with the migration.

here it is: https://issues.redhat.com/browse/CRT-1036 ;)

Reply:
It seems that you changed
```
if condition.IsTrueWithReason(userSignup.Status.Conditions, toolchainv1alpha1.UserSignupComplete, "") 
```
to
```
if state == toolchainv1alpha1.UserSignupStateLabelValueApproved ||
	state == toolchainv1alpha1.UserSignupStateLabelValueDeactivated ||
	state == toolchainv1alpha1.UserSignupStateLabelValueBanned
```
But technically it may happen that there is a UserSIgnup with, let's say, "approved" label but without the corresponding MUR yet.
UserSignup controller adds the "approved" label first then trying to provision a new MUR. If between setting the label and provisioning a MUR we deploy a new host operator with this migration then we will count such UserSignups twice. Once when we reconcile the UserSignup without the counter label (but with "approved" already set) and the second time when the MUR provisioning resumes (you increment the counter when creating a MUR).

Initially I thought that it would be safer initiate the counter label (when missing) only during MUR ensuring. But we would miss deactivated and banned users in this case. We could improve it and cover all the cases but I guess it's really an edge case and we could leave it as is. In worst case scenario we could miscount one or two accounts during the initial migration which is not a big deal

Reply:
yes, I guess that the situation that you described is really an edge-case and that it may happen that we could count one or two accounts twice, which is not a big deal. 
This migration happens only once and the metric will have thousands of activations == 1, so one or two extra counts is fine.
I would keep it as it is.